contrib/cmake/build3rdparty.bash

```diff
@@ -1,17 +1,88 @@
-#!/bin/bash
-# The usual way to use this file is to make a new folder 3rdParty next to your sdk folder, copy this script to it, and run it there.
-# Then building via cmake can refer to these libraries.
-
-git clone https://github.com/Microsoft/vcpkg.git
-cd vcpkg
-./bootstrap-vcpkg.sh
-./vcpkg install --triplet x64-linux zlib
-./vcpkg install --triplet x64-linux cryptopp
-./vcpkg install --triplet x64-linux openssl
-./vcpkg install --triplet x64-linux c-ares
-./vcpkg install --triplet x64-linux curl    
-./vcpkg install --triplet x64-linux sqlite3
-./vcpkg install --triplet x64-linux gtest
-#./vcpkg install --triplet x64-linux libsodium
+# This script is for getting and building the 3rd party libraries that the MEGA SDK uses (some are optional, and some are only needed only by MEGA apps too)
+# 
+# Your 3rdParty library builds should be outside the SDK repo.  We are moving to use vcpkg to build most of them. You can start it like this:
+#
+# mkdir 3rdParty
+# cd 3rdParty
+# git clone https://github.com/Microsoft/vcpkg.git
+# cd vcpkg
+# .\bootstrap-vcpkg.bat
+#
+# Note that our current toolset for XP compatibility (which we had but we are deprecating), is VS2015 with v140 or v140_xp.  
+# The triplet files ending -mega set v140, please adjust to your requirements.
+#
+# Another thing you might want to consider for developing, is turning off checked iterators in MSVC builds, since those extra tests can cause big delays
+# in debug (eg. deleting the node tree on logout etc).  You can do that by setting set(VCPKG_CXX_FLAGS "${VCPKG_CXX_FLAGS} -D_ITERATOR_DEBUG_LEVEL=0")
+# in your triplet file, though every library used must have that same setting.
+# 
+# Another way is to edit the STL headers first, which can ensure that nested builds get the right flag.
+# 
+# Copy the folders in contrib/cmake/vcpkg_extra_ports to your 3rdParty/vcpkg/ports folder.
+# Copy the folders in contrib/cmake/vcpkg_extra_triplets to your 3rdParty/vcpkg/triplets folder.
+# 
+# Comment out any libraries that you won't use.
+# If using pdfium, follow the instructions below to get the source code
+#
+# On Mac, go to ports/libraw/CONTROL and comment out the freeglut dependency
+# 
+# Copy this script to your 3rdParty/vcpkg folder, and run it with the desired triplet as the parameter.  (usually x64-windows-mega or x86-windows-mega)
```

Comment:
coyping it to the vcpkg folder is not required. x64-windows... does not make sense in this bash script, doesn't it?
We can have the ports & triplet linked from the location of this script (build3rpardy.bash) automatically. Sth like:
```
DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
ln -sf $DIR/vcpkg_extra_ports/* ports/
ln -sf $DIR/vcpkg_extra_triplets/* triplets/
```

Reply:
That's a very interesting idea - it would be great for the script to automate as much as possible.  It looks like this assumes that the script would stay in the contrib/cmake folder, and be invoked from the 3rdparty/vcpkg folder.   I've been copying it to the target folder and running from there - but I think that would be fine.  I'll make the update.

Reply:
Also we should think about adding an x64-linux-mega triplet file to the extras folder, which could select which libraries will be built for static linking in a standard build, similar to the x64-windows-mega triplet file.